src/messaging/js/composeui-messaging-client/src/client/MessageRouterClient.ts

```diff
@@ -83,11 +83,16 @@ export class MessageRouterClient implements MessageRouter {
         const subscription = topic.subscribe(subscriber);
 
         if (needsSubscription) {
-            await this.sendMessage<messages.SubscribeMessage>(
-                {
-                    type: "Subscribe",
-                    topic: topicName
-                });
+            try {
+                await this.sendRequest<messages.SubscribeMessage, messages.SubscribeResponse>(
+                    {
+                        requestId: this.getRequestId(),
+                        type: "Subscribe",
+                        topic: topicName
+                    });
+            } catch (error) {
```

Comment:
What's the point of just throwing in a catch block?

Reply:
It's also rewriting the exception stack to root here, is that intentional?

Reply:
That's not a bad idea, the error object is created by `sendRequestAsync`, but it makes more sense to see the current method in the stack trace, as if the error was thrown here.

Reply:
My original idea was that. I will remove the try catch block then from the related calls I modified.